samples/extensions/debug_utils/debug_utils.cpp

```diff
@@ -0,0 +1,1135 @@
+		// Depth attachment
+		create_attachment(depth_format, VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT, &offscreen.depth);
+
+		// Set up separate renderpass with references to the colorand depth attachments
+		std::array<VkAttachmentDescription, 3> attachment_descriptions = {};
+
+		// Init attachment properties
+		for (uint32_t i = 0; i < 3; ++i)
+		{
+			attachment_descriptions[i].samples        = VK_SAMPLE_COUNT_1_BIT;
+			attachment_descriptions[i].loadOp         = VK_ATTACHMENT_LOAD_OP_CLEAR;
+			attachment_descriptions[i].storeOp        = VK_ATTACHMENT_STORE_OP_STORE;
+			attachment_descriptions[i].stencilLoadOp  = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+			attachment_descriptions[i].stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+			if (i == 2)
+			{
+				attachment_descriptions[i].initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+				attachment_descriptions[i].finalLayout   = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+			}
+			else
+			{
+				attachment_descriptions[i].initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+				attachment_descriptions[i].finalLayout   = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+			}
+		}
+
+		// Formats
+		attachment_descriptions[0].format = offscreen.color[0].format;
+		attachment_descriptions[1].format = offscreen.color[1].format;
+		attachment_descriptions[2].format = offscreen.depth.format;
+
+		std::vector<VkAttachmentReference> color_references;
+		color_references.push_back({0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL});
+		color_references.push_back({1, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL});
+
+		VkAttachmentReference depth_reference = {};
+		depth_reference.attachment            = 2;
+		depth_reference.layout                = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
+
+		VkSubpassDescription subpass    = {};
+		subpass.pipelineBindPoint       = VK_PIPELINE_BIND_POINT_GRAPHICS;
+		subpass.pColorAttachments       = color_references.data();
+		subpass.colorAttachmentCount    = 2;
+		subpass.pDepthStencilAttachment = &depth_reference;
+
+		// Use subpass dependencies for attachment layput transitions
+		std::array<VkSubpassDependency, 2> dependencies;
+
+		dependencies[0].srcSubpass      = VK_SUBPASS_EXTERNAL;
+		dependencies[0].dstSubpass      = 0;
+		dependencies[0].srcStageMask    = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
+		dependencies[0].dstStageMask    = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+		dependencies[0].srcAccessMask   = VK_ACCESS_MEMORY_READ_BIT;
+		dependencies[0].dstAccessMask   = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+		dependencies[0].dependencyFlags = VK_DEPENDENCY_BY_REGION_BIT;
+
+		dependencies[1].srcSubpass      = 0;
+		dependencies[1].dstSubpass      = VK_SUBPASS_EXTERNAL;
+		dependencies[1].srcStageMask    = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+		dependencies[1].dstStageMask    = VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT;
```

Comment:
The intention is to read the output of this pass in the fragment shader in the following pass, so `VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT`?